### DISCUSSION TITLE
Altering the postions of the options visible in top toolbar

### DISCUSSION THREAD

**Discussion by sowjanyagunupuru** [eclipse-theia/theia#14250] [Q&A] [answered]
I have a menu bar icon in my left panel 
On click on that menu I want to show the options( files, view,go ,deploy .. all these)
Which are in the top panel i.e toolbar  as a popup or small toast 

**Requested note 
The functionality must not change
Only position of those options to be changed 

Clearly I shown my requirement in the attached image

![IMG_20241004_215400](https://github.com/user-attachments/assets/1e082347-456c-418e-93e4-c18c2aa18b7a)

**Comment by msujew** [comment]
Hey @sowjanyagunupuru,

have you tried setting the `window.menuBarVisibility` preference to `compact`? I believe that should achieve what you're trying to do.

**Reply by sowjanyagunupuru** [reply]
Yes I tried that setting to compact 

But on setting the preference to compact 
The top panel i.e toolbar is not visible 

But I want the toolbar to be visible and the menu options to be in compact 
Because I added my additional items in the top-panel
So I want to move these view go terminal action .... Options to compact format

**Reply by msujew** [reply]
I see. Currently, the top menu and the compact menu option are mutually exclusive. You will likely need to create a more extensive override of the `ApplicationShell` service to build what you want to achieve.

**Reply by sowjanyagunupuru** [reply]
Can I make the changes to application shell
Like even it is in compact option 
The top panel must be visible

toppanel.hide()

Or by  modifying the 
HiddenPreferences = ['hidden'];

Can u kindly suggest me the way to get my requested changes to work well

**Reply by msujew** [reply, marked as answer]
Please see:

https://github.com/eclipse-theia/theia/blob/ac0c246fbcd439c259b4778e41199708d4354899/packages/core/src/browser/shell/application-shell.ts#L304-L313

and 

https://github.com/eclipse-theia/theia/blob/ac0c246fbcd439c259b4778e41199708d4354899/packages/core/src/browser/shell/application-shell.ts#L379-L382

**Reply by sowjanyagunupuru** [reply]
Thank you for ur immediate responses 😊 
I will update once I go through this

**Reply by sowjanyagunupuru** [reply]
Everything is working according my requirement.
Thank you for ur support in providing us the way to get the solution


Here the additional change i want in this is
Either I want to show only the 'compact' option in the dropdown options of Window.Menu Bar Visibility and to remove the rest options like classic , visible, hidden 

And the main important requirement is that  the 'compact' to be set as the default layout preference instead of classic
![IMG_20241008_123517](https://github.com/user-attachments/assets/372f1f03-9d1e-48a4-9b67-e9dcb6bb47d7)

**Reply by msujew** [reply]
You can change the default for specific preferences, see:

https://github.com/eclipse-theia/theia-blueprint/blob/ed203687bad53fdbd22012926978ac40d7d8cd23/applications/electron/package.json#L22-L31

As for removing parts of the preference: You can change this object to achieve what you want

https://github.com/eclipse-theia/theia/blob/b388d4cf088c5efb7e2f41d1973db5be3d004868/packages/core/src/browser/core-preferences.ts#L94-L107

**Reply by sowjanyagunupuru** [reply]
To change the default value 

I kept as shown in the image  in package.json of browser-app, even though my default value is 'classic' only 
May I know Where I went wrong 
I want to make the default as 'compact'

![IMG_20241008_151116](https://github.com/user-attachments/assets/6dcdcdee-79b7-4cc1-8e8d-4d713a9845a2)

**Reply by msujew** [reply]
Have you already changed your preferences once? If so, the change is stored in your `settings.json` file. In that case, the default will not take effect, since you already changed the preference.

**Reply by sowjanyagunupuru** [reply]
Hello @msujew 

Initially before changing any preferences,
Removed .theia file too from c-drive
At the first attempt of yarn build and start
 
I got the ui opened with the 'compact' option ( in drop-down window.menubar visibility - the shown selected option is compact but the default is for classic)

And also then toppanel is invisible 
But the menubar icon to show menu options is not visible 

This is the issue which is taking more time for me to understand to solve 

This issue is when I am opening with a workspace 

If I am opening with users(user level)
That's perfectly working in compact 

Hope u understood my issue 
Solution for this is very much needed

**Reply by sowjanyagunupuru** [reply]
Please have a look to the attached image
At first opening of the interface after deleting settings.json in .theia in c drive

The option shown is compact
But there is no menu bar icon in the left panel
And also no top panel 
![IMG_20241009_094735](https://github.com/user-attachments/assets/ab989e4f-5cbc-4252-9cc6-5c835bb7d492)


Means here it's showing the compact layout but the menu options are not added at left panel 

And also have a look to the drop-down in below image 
Though it's showing the compact as selected options at the beginning before changing preference but the default value is set to classic 
![IMG_20241009_095111](https://github.com/user-attachments/assets/597579ca-8c40-4780-8af8-c2325350afba)

**Reply by msujew** [reply]
Can you provide a reproducible example? It's hard to tell what is wrong from the information you've shared.

**Reply by sowjanyagunupuru** [reply]
In the image 
![17284677392338131743655098918457](https://github.com/user-attachments/assets/2400d332-052e-4341-b11a-714ce0b4e50b)

The default is classic 
But I want to make it as compact 

For that I want to override the properties
![IMG_20241009_152807](https://github.com/user-attachments/assets/485ebb47-d271-4ff3-aff7-ea18997bd123)


Here  in the cire-preferences.ts 
The enum:['compact']
Except compact no other option to be in enum of window.menuBarVisibility

How can I override that 
Can u please suggest the way  to make the window:Menu Bar Visibility has only one option - 'compact'

**Reply by msujew** [reply]
Can you click on the `gear` icon next to the preference and reset the preference? It should reset itself to `compact`. The `default` label likely isn't correctly calculated.

**Reply by sowjanyagunupuru** [reply]
Yes
The drop-down menu currently includes all options: Classic, Visible, Hidden, and Compact. However, I would like only the Compact option to be visible to users in the menu bar visibility settings. Users should only have the ability to select Compact, restricting them from seeing or choosing any other visibility options.

**Reply by msujew** [reply]
You can set the `hidden` property of the preference to `true` to prevent the preference from showing up in the settings view.

**Reply by sowjanyagunupuru** [reply]
I am sry @msujew 
Actually my requirement exactly is to make the menu bar visibility drop down as read only 
Or to eliminate, remove the options except the compact from the drop down

**Reply by msujew** [reply]
> Or to eliminate, remove the options except the compact from the drop down

Then you can just set the `enum` values to just one value like `['compact']`.

**Reply by sowjanyagunupuru** [reply]
Hello @msujew 

In theia I changed the File icon(theia) 
To seti Seti (Vishal Studio Code)

Here I want some changes in it
Like 
1) i want to decrease the icon size
2) want to include space between the filename and icon as shown in below image 
![IMG_20241011_123104](https://github.com/user-attachments/assets/ae4ba1c0-535a-4a91-8954-441dbca6854d)


Kindly may I get the suggestion how can I make those my required changes

**Reply by msujew** [reply]
Take a look at the HTML classes in the devtools of your browser and you should be able to achieve this via CSS.
